helm/prometheus-agent/charts/prometheus-agent/templates/prometheus.yaml

```diff
@@ -22,6 +22,13 @@ spec:
       name: http-web
       protocol: TCP
     {{- if .Values.watchdog.enabled }}
+    lifecycle:                                                                                                                                                                                                                                                
+      preStop:                                                                                                                                                                                                                                                
+        exec:                                                                                                                  
+          command:                                                                                                             
+          - /bin/sh                                                                                                            
+          - -c                                                                                                                 
+          - rm -f /prometheus/wal/*                                                                                            
```

Comment:
I'm not sure globbing will work here. I would have done `rm -rf /prometheus/wal`.
Let's see how it passes the night test :smile: